released/discovery_center/mission_4356/main.tf

```diff
@@ -0,0 +1,153 @@
+}
+
+###############################################################################################
+# Creation of subaccount
+###############################################################################################
+resource "btp_subaccount" "project" {
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  region    = lower(var.region)
+}
+
+###############################################################################################
+# Assignment of users as sub account administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Assignment of users as sub account service administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+######################################################################
+# Creation of Cloud Foundry environment
+######################################################################
+module "cloudfoundry_environment" {
+  source                    = "../../modules/environment/cloudfoundry/envinstance_cf"
+  subaccount_id             = btp_subaccount.project.id
+  instance_name             = local.project_subaccount_cf_org
+  plan_name                 = "standard"
+  cf_org_name               = local.project_subaccount_cf_org
+  cf_org_auditors           = var.cf_org_auditors
+  cf_org_managers           = var.cf_org_managers
+  cf_org_billing_managers   = var.cf_org_billing_managers
+}
+
+# ######################################################################
+# # Creation of Cloud Foundry space
+# ######################################################################
+# module "cloudfoundry_space" {
+#   source              = "../../modules/environment/cloudfoundry/space_cf"
+#   cf_org_id           = module.cloudfoundry_environment.cf_org_id
+#   name                = var.cf_space_name
+#   cf_space_managers   = var.cf_space_managers
+#   cf_space_developers = var.cf_space_developers
+#   cf_space_auditors   = var.cf_space_auditors
+# }
+
+######################################################################
+# Add "sleep" resource for generic purposes
+######################################################################
+resource "time_sleep" "wait_a_few_seconds" {
```

Comment:
Sleep resource is not needed, but you must put the data sources and the corresponding resources into modules to allow a decent dependency management